src/modules-lua/noit/module/ssh2.lua

```diff
@@ -48,23 +48,23 @@ function onload(image)
     <parameter name="method_kex"
                required="optional"
                default="diffie-hellman-group14-sha1"
-               allowed="^diffie-hellman-(?:group1-sha1|group14-sha1|group-exchange-sha1)$">The key exchange method to use.</parameter>
```

Comment:
You've dropped group-exchange-sha1 ?

Reply:
Yes, because it doesn't work.  If you think it should work, I can have another look, but I'm fairly certain I tested it and it failed.

Reply:
From my understanding of RFC 4419, if we want to do `diffie-hellman-group-exchange-*` we have to construct a different kexinit packet that lists our min/max/preferred group size and let the server choose the group.  I wasn't prepared to go down that particular rabbit hole.